doc/developer/platform/ux.md

```diff
@@ -285,31 +285,152 @@ The HTTPS server exposes two APIs:
 
   * `POST /query` executes a batch of SQL statements.
 
-    The request body specifies the SQL string to execute. The request may contain
-    multiple SQL statements separated by semicolons. The statements are submitted
-    for processing as if they were supplied via the [Simple Query][simple-query]
-    flow of the PostgreSQL protocol (i.e., "the statements are executed as a
-    single transaction unless explicit transaction control commands are included
-    to force a different behavior").
-
-    The response is a JSON array containing the results of each query in the
-    request.
+    The request body is JSON containing a key, `sql`, which specifies the SQL
+    string to execute.
+    - The request may contain multiple SQL statements separated by semicolons.
+    - The statements are submitted for processing as if they were supplied via
+    the [Simple Query][simple-query] flow of the PostgreSQL protocol (i.e., "the
+    statements are executed as a single transaction unless explicit transaction
+    control commands are included to force a different behavior").
+    - We abort the request upon encountering the first error.
```

Comment:
Errr, this should be `s/request/transaction/`; because we treat all of these statements as being bundled in the same transaction, I believe the erroneous `commit` would invalidate the transaction and the insert would not commit. I'll dig into this .

Reply:
Ok, amended this with my corrected understanding of how transaction processing works.